fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+        call ESMF_ConfigGetAttribute(config=CF, value=dx,      label ='dx:',     rc=rc)
+        call ESMF_ConfigGetAttribute(config=CF, value=dy,      label ='dy:',     rc=rc)
+        if (mype == 0) then
+          print *,'cen_lon=',cen_lon,' cen_lat=',cen_lat
+          print *,'stdlat1=',stdlat1,' stdlat2=',stdlat2
+          print *,'lon1=',lon1,' lat1=',lat1
+          print *,'nx=',imo, ' ny=',jmo
+          print *,'dx=',dx,' dy=',dy
+        endif
+      endif ! output_grid
+
+    endif ! quilting
+!
+    call ESMF_ConfigGetAttribute(config=CF, value=dt_atmos, label ='dt_atmos:',   rc=rc)
+    call ESMF_ConfigGetAttribute(config=CF, value=nfhmax,   label ='nhours_fcst:',rc=rc)
+    if(mype == 0) print *,'af nems config,dt_atmos=',dt_atmos,'nfhmax=',nfhmax
+    call ESMF_TimeIntervalSet(timeStep,s=dt_atmos,rc=rc)
+    call ESMF_ClockSet(clock_fv3,timeStep=timeStep, rc=rc)
+!
+!------------------------------------------------------------------------
+! may need to set currTime for restart
+!
+    call ESMF_ClockGet(clock_fv3, currTIME=CurrTime,  StartTime=startTime,    &
+                       RunDuration=RunDuration, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+    StopTime = startTime + RunDuration
+
+! *** read restart time from restart file
+    do i=751,899
+       inquire(i, opened=OPENED)
+       if(.not. OPENED)then
+         io_unit = i
+         exit
+       endif
+    enddo
+!
+    date = 0 ; date_init = 0
+    force_date_from_configure = .true.
+!
+    open(unit=io_unit, file=trim('INPUT/coupler.res'),status="old",err=998 )
+    read (io_unit,*,err=999) calendar_type
+    read (io_unit,*) date_init
+    read (io_unit,*) date
+    close(io_unit)
+    force_date_from_configure = .false.
+!
+    if(date(1) == 0 .and. date_init(1) /= 0) date = date_init
+    if(mype == 0) print *,'bf clock_fv3,date=',date,'date_init=',date_init
+
+    call ESMF_VMbroadcast(vm, date, 6, 0)
+    call ESMF_TimeSet(time=CurrTime,yy=date(1),mm=date(2),dd=date(3),h=date(4), &
+                      m=date(5),s=date(6),rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+999 continue
+998 continue
+!    if(mype==0) print *,'final date =',date,'date_init=',date_init
+
+!reset CurrTime in clock
+    call ESMF_ClockSet(clock_fv3, currTIME=CurrTime, startTime=startTime,  &
+                       stopTime=stopTime, timeStep=timeStep, rc=rc)
+
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+ 
+    !
+    !Under NUOPC, the EARTH driver clock is a separate instance from the
+    ! - fv3 clock. However, the fv3 clock may have been reset from restart
+    ! - therefore the EARTH driver clock must also be adjusted.
+    ! - Affected: currTime, timeStep
+    call ESMF_ClockGet(clock, timeStep=earthStep, rc=RC)
```

Comment:
This section has the atmosphere model update the Earth clock based on it's own restart status.  Restarts need to be handled across all the components - so an Earth driver level configuration is needed.  This information can be passed down to each component to let them know that it is a restart.  Otherwise, how would you handle a restart in the case of a DATM-MOM6-CICE run, since FV3 will not be present to "reset" the Earth clock.